packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -75,19 +132,24 @@ export class Application extends ApplicationBase {
 
   public readonly applicationArn: string;
   public readonly applicationId: string;
+  private readonly application: CfnApplication
 
   constructor(scope: Construct, id: string, props: ApplicationProps) {
     super(scope, id);
 
     this.validateApplicationProps(props);
 
-    const application = new CfnApplication(this, 'Resource', {
+    this.application = new CfnApplication(this, 'Resource', {
       name: props.applicationName,
       description: props.description,
     });
 
-    this.applicationArn = application.attrArn;
-    this.applicationId = application.attrId;
+    this.applicationArn = this.application.attrArn;
+    this.applicationId = this.application.attrId;
+  }
+
+  protected generateUniqueHash(value: string): string {
+    return hashValues(cdk.Names.nodeUniqueId(this.application.node), value);
```

Comment:
Can't we just save the value of `dk.Names.nodeUniqueId(this.application.node)` as a private readonly field? Will allow you to roll back the change of having `application` as a field, plus will not recompute this value for multiple calls to `associate*()`.

Reply:
good call.